compiler/qsc_fir/src/fir.rs

```diff
@@ -894,6 +894,9 @@ impl Display for SpecDecl {
     }
 }
 
+/// An execution graph represented by a reference counted vector of nodes.
+pub type ExecGraph = Rc<[ExecGraphNode]>;
```

Comment:
I was worried when I saw all the "cloning" of a graph above until I got to this. Maybe the type alias could reflect that it's a reference count (e.g. `RcExecGraph` or something).

Reply:
When you are in the editor with the benefit of rust-analyzer it shows the definition and the comment, does that help?
<img width="625" alt="image" src="https://github.com/user-attachments/assets/e978b4ed-24ac-4da2-b1fd-02598f6c262b">